src/sysinfo/nvapi_adapter.cpp

```diff
@@ -147,7 +157,51 @@ namespace dxvk {
         return 0;
     }
 
-    bool NvapiAdapter::isVkDeviceExtensionSupported(const std::string name) { // NOLINT(performance-unnecessary-value-param)
+    bool NvapiAdapter::GetLUID(LUID *luid) const {
+        if (m_deviceIdProperties.deviceLUIDValid) {
+            memcpy(luid, &m_deviceIdProperties.deviceLUID, sizeof(*luid));
+            return true;
+        }
+        return false;
+    }
+
+    bool NvapiAdapter::GetArchitectureId(NV_GPU_ARCHITECTURE_ID *archId) const {
+        // This function only makes sense on NVIDIA GPUs, bail out early if a
+        // non-NVIDIA GPU is detected.
+        if (m_deviceProperties.vendorID != 0x10de)
```

Comment:
What about moving this test into its own method and use it in the calling method (lNvAPI_GPU_GetArchInfo)? Like:
```c++
if (!adapter-IsNvidiaGpu())
  return NvidiaDeviceNotFound(n);
```
That way we could just return NV_GPU_ARCHITECTURE_ID in this here by declaring it like
```c++
NV_GPU_ARCHITECTURE_ID NvapiAdapter::GetArchitectureId() const
```

Reply:
Fixed.